src/System.IO.Pipelines.Networking.Tls/Internal/OpenSsl/Interop.cs

```diff
@@ -0,0 +1,451 @@
+        public delegate AlpnStatus alpn_cb(IntPtr ssl, out byte* selProto, out byte selProtoLen, byte* inProtos, int inProtosLen, IntPtr arg);
+
+        public static IntPtr NewServerContext(VerifyMode verifyMode) => IsWindows ? WindowsLib.NewServerContext(verifyMode) : IsOsx ? OsxLib.NewServerContext(verifyMode) : UnixLib.NewServerContext(verifyMode);
+        public static IntPtr NewClientContext(VerifyMode verifyMode) => IsWindows ? WindowsLib.NewClientContext(verifyMode) : IsOsx ? OsxLib.NewClientContext(verifyMode) : UnixLib.NewClientContext(verifyMode);
+
+        public static void SSL_set_bio(IntPtr ssl, InteropBio.BioHandle readBio, InteropBio.BioHandle writeBio)
+        {
+            if (IsWindows)
+            {
+                WindowsLib.SSL_set_bio(ssl, readBio.Handle, writeBio.Handle);
+            }
+            else if (IsOsx)
+            {
+                OsxLib.SSL_set_bio(ssl, readBio.Handle, writeBio.Handle);
+            }
+            else
+            {
+                UnixLib.SSL_set_bio(ssl, readBio.Handle, writeBio.Handle);
+            }
+        }
+        public static void SSL_set_connect_state(IntPtr ssl)
+        {
+            if (IsWindows)
+            {
+                WindowsLib.SSL_set_connect_state(ssl);
+            }
+            else if (IsOsx)
+            {
+                OsxLib.SSL_set_connect_state(ssl);
+            }
+            else
+            {
+                UnixLib.SSL_set_connect_state(ssl);
+            }
+        }
+        public static void SSL_set_accept_state(IntPtr ssl)
+        {
+            if (IsWindows)
+            {
+                WindowsLib.SSL_set_accept_state(ssl);
+            }
+            else if (IsOsx)
+            {
+                OsxLib.SSL_set_accept_state(ssl);
+            }
+            else
+            {
+                UnixLib.SSL_set_accept_state(ssl);
+            }
+        }
+        public static int SSL_write(IntPtr ssl, void* buf, int len) => IsWindows ? WindowsLib.SSL_write(ssl, buf, len) : IsOsx ? OsxLib.SSL_write(ssl, buf, len) : UnixLib.SSL_write(ssl, buf, len);
+        public static int SSL_read(IntPtr ssl, void* buf, int len) => IsWindows ? WindowsLib.SSL_read(ssl, buf, len) : IsOsx ? OsxLib.SSL_read(ssl, buf, len) : UnixLib.SSL_read(ssl, buf, len);
+        public static int SSL_do_handshake(IntPtr ssl) => IsWindows ? WindowsLib.SSL_do_handshake(ssl) : IsOsx ? OsxLib.SSL_do_handshake(ssl) : UnixLib.SSL_do_handshake(ssl);
+        public static SslErrorCodes SSL_get_error(IntPtr ssl, int errorIndetity) => IsWindows ? WindowsLib.SSL_get_error(ssl, errorIndetity) : IsOsx ? OsxLib.SSL_get_error(ssl, errorIndetity) : UnixLib.SSL_get_error(ssl, errorIndetity);
+        public static int SetKeys(IntPtr ctx, IntPtr certificates, IntPtr privateKey) => IsWindows ? WindowsLib.SetKeys(ctx, certificates, privateKey) : IsOsx ? OsxLib.SetKeys(ctx, certificates, privateKey) : UnixLib.SetKeys(ctx, certificates, privateKey);
```

Comment:
because if there is no return value you can't use a single-line-double-ternary as far as I am aware, maybe you can